lib/resty/iprepd.lua

```diff
@@ -0,0 +1,76 @@
+local cjson = require('cjson')
+local lrucache = require('resty.lrucache')
+local http = require('resty.http')
+
+function fatal_error(error_msg)
+  ngx.log(ngx.ERR, error_msg)
+  os.exit(1)
+end
+
+local _M = {}
+local mt = { __index = _M }
+
+function _M.new(options)
+  local iprepd_url = options.url or 'http://localhost:8080/'
+  local cache_ttl = options.cache_ttl or 30
+  local iprepd_threshold = options.threshold
+  if not iprepd_threshold then
+    fatal_error('Need to pass in a threshold')
+  end
+  local iprepd_api_key = options.api_key
```

Comment:
you might be able to simplify this a bit

`local iprepd_api_key = options.api_key or fatal_error('Need to pass in an api_key')`